packages/hardhat-core/test/internal/hardhat-network/helpers/useProvider.ts

```diff
@@ -115,6 +122,79 @@ export function useProvider({
         this.server.getProvider()
       );
     }
+
+    if (rethnetBinary !== undefined) {
+      this.rethnetProcess = spawn(normalize(rethnetBinary), ["node", "-vv"]);
+
+      let stdout = "";
+      this.rethnetProcess.stdout.on("data", (data: any) => {
+        stdout += data.toString();
+        console.log(`rethnet subprocess ${this.rethnetProcess.pid}: ${data}`);
+      });
+
+      let stderr = "";
+      this.rethnetProcess.stderr.on("data", (data: any) => {
+        stderr += data.toString();
+        console.log(`rethnet subprocess ${this.rethnetProcess.pid}: ${data}`);
+      });
+
+      function outputForError() {
+        return `stdout:\n${stdout}\nstderr:\n${stderr}`;
+      }
+
+      const wait = new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
There is a sleep helper you can use, in the `sleep.ts` module in the same directory.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4095/commits/368202e32b4226e46905d22cb08899d4f2cbab97